### PR TITLE
Report back the detected version of the remote Elasticsearch

### PR DESCRIPTION
**Pull request by ph**
## What is the problem this PR solves?

Previously the system was returning  an error with only an `unsupported version` message.

```
Fleet Server - Error - failed version compatibility check with elasticsearch: unsupported version'
```

Instead, we will return

```
Fleet Server - Error - failed version compatibility check with elasticsearch (Agent: 8.2.0, Elasticsearch: 7.17.0): unsupported version'
```

## How does this PR solve the problem?

// Explain HOW you solved the problem in your code. It is possible that during PR reviews these changes and then this section should be updated.

## How to test this PR locally

Start Fleet server with an older version of Elasticsearch.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates #1210

### CODE REVIEW

**Review by blakerouse — APPROVED** [review]
NIce! Looks good.

**Comment by ph** [comment]
taking a look with the check, seems some go deps.

File: go.sum [hunk 1]
```diff
@@ -1827,8 +1559,6 @@ sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.14/go.mod h1:LEScyz
 sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.15/go.mod h1:LEScyzhFmoF5pso/YSeBstl57mOzx9xlU9n85RGrDQg=
 sigs.k8s.io/structured-merge-diff/v4 v4.0.2/go.mod h1:bJZC9H9iH24zzfZ/41RGcq60oK1F7G282QMXDPYydCw=
 sigs.k8s.io/structured-merge-diff/v4 v4.0.3/go.mod h1:bJZC9H9iH24zzfZ/41RGcq60oK1F7G282QMXDPYydCw=
-sigs.k8s.io/structured-merge-diff/v4 v4.1.0 h1:C4r9BgJ98vrKnnVCjwCSXcWjWe0NKcUQkmzDXZXGwH8=
 sigs.k8s.io/structured-merge-diff/v4 v4.1.0/go.mod h1:bJZC9H9iH24zzfZ/41RGcq60oK1F7G282QMXDPYydCw=
 sigs.k8s.io/yaml v1.1.0/go.mod h1:UJmg0vDUVViEyp3mgSv9WPwZCDxu4rQW1olrI1uml+o=
-sigs.k8s.io/yaml v1.2.0 h1:kr/MCeFWJWTwyaHoR9c8EjH9OumOmoF9YGiZd7lFm/Q=
 sigs.k8s.io/yaml v1.2.0/go.mod h1:yfXDCHCao9+ENCvLSE62v9VSji2MKu5jeNfTrofGhJc=
```
Comment (ph) [inline go.sum:1564]: use the wrong key in magit. :(

File: cmd/fleet/main.go [hunk 2]
```diff
@@ -785,8 +787,11 @@ func (f *FleetServer) runSubsystems(ctx context.Context, cfg *config.Config, g *
 	esCli := bulker.Client()
 
 	// Check version compatibility with Elasticsearch
-	err = ver.CheckCompatibility(ctx, esCli, f.bi.Version)
+	remoteVersion, err := ver.CheckCompatibility(ctx, esCli, f.bi.Version)
 	if err != nil {
+		if len(remoteVersion) != 0 {
+			return fmt.Errorf("failed version compatibility check with elasticsearch (Agent: %s, Elasticsearch: %s): %w", f.bi.Version, remoteVersion, err)
```
Comment (ph) [inline cmd/fleet/main.go:793]: Note: I understand we are in Fleet-Server, but the user only interacts with agent.

**Pull request merged**